stdlib/email/src/main/ballerina/src/email/listener_endpoint.bal

```diff
@@ -0,0 +1,196 @@
+    public function __gracefulStop() returns error? {
+        check self.stop();
+    }
+
+    function start() returns error? {
+        var scheduler = self.config.cronExpression;
+        if (scheduler is string) {
+            task:AppointmentConfiguration config = {appointmentDetails: scheduler};
+            self.appointment = new(config);
+        } else {
+            task:TimerConfiguration config = {intervalInMillis: self.config.pollingInterval, initialDelayInMillis: 100};
+            self.appointment = new (config);
+        }
+        var appointment = self.appointment;
+        if (appointment is task:Scheduler) {
+            check appointment.attach(appointmentService, self);
+            check appointment.start();
+        }
+        log:printInfo("Listening to remote server at " + self.config.host + "...");
+    }
+
+    function stop() returns error? {
+        var appointment = self.appointment;
+        if (appointment is task:Scheduler) {
+            check appointment.stop();
+        }
+        log:printInfo("Stopped listening to remote server at " + self.config.host);
+    }
+
+    # Polls the email server enspoint.
+    # ```ballerina
+    # email:Error? result = emailListener.poll();
+    # ```
+    #
+    # + return - () or else error upon failure to poll the listener
+    public function poll() returns error? {
+        return poll(self.config);
+    }
+
+    # Registers for the Email service.
+    # ```ballerina
+    # email:Error? result = emailListener.register(helloService, hello);
+    # ```
+    #
+    # + emailService - Type descriptor of the service
+    # + name - Service name
+    # + return - `()` or else a `email:Error` upon failure to register the service
+    public function register(service emailService, string? name) returns error? {
+        error? response = ();
+        handle|error result = register(self, self.config,  emailService);
+        if(result is handle){
+            self.config.serverConnector = result;
+        } else {
+            response = result;
+        }
+        return response;
+    }
+};
+
+service appointmentService = service {
+    resource function onTrigger(Listener l) {
+        var result = l.poll();
+        if (result is error) {
+            log:printError("Error while executing poll function", result);
+        }
+    }
+};
+
+# Configuration for Email listener endpoint.
+#
+# + host - Email server host
+# + username - Email server access username
+# + password - Email server access password
+# + protocol - Email server access protocol, "IMAP" or "POP"
+# + protocolConfig - POP3 or IMAP4 protocol configuration
+# + pollingInterval - Periodic time interval to check new update
+# + cronExpression - Cron expression to check new update
+# + serverConnector - Server connector for service
+public type ListenerConfig record {|
+    string host = "127.0.0.1";
+    string username = "";
+    string password = "";
```

Comment:
Shall we make the host, username, and password mandatory? since it is required to login

Reply:
Made them as mandatory fields.